### PR TITLE
feat(challenge): add update-pubkey action for BIP-322 agents (closes #399)

### LINKED ISSUE #399: btcPublicKey stored as empty string for BIP-322 wallets; add update-pubkey challenge
## Bug: btcPublicKey silently stored as empty string

**Reported by**: User stuck on this issue

During registration, `verifyBitcoinSignature()` in `lib/bitcoin-verify.ts` returns `publicKey: ""` for BIP-322 signatures (bc1q/bc1p wallets). BIP-322 verification confirms the address owns the key but cannot recover the public key from the signature (unlike BIP-137 which does key recovery).

The empty string is then stored directly into the AgentRecord at `app/api/register/route.ts:614` with no validation or warning:

```ts
btcPublicKey: btcResult.publicKey, // "" for BIP-322 wallets
```

**Impact**: The cryptographic tie between L1 (Bitcoin) and L2 (Stacks) is broken — the agent has no usable public key on record. This affects:
- Nostr npub derivation (returns null for empty pubkey)
- Any future feature that relies on `btcPublicKey` for identity verification
- The fundamental L1↔L2 binding that the platform's trust model depends on

### 1. Stop silently accepting empty pubkeys (bug fix)

During registration, if `btcResult.publicKey` is empty:
- Log a warning
- Still allow registration (the address is verified)
- Include a flag/note in the response indicating the pubkey needs to be set
- Consider requiring the agent provide their pubkey explicitly if BIP-322 was used

### 2. Add `update-pubkey` challenge action (feature)

Add an `update-pubkey` action to the challenge/response system (`lib/challenge.ts`) so agents can fix a missing or empty `btcPublicKey`:

- Agent requests challenge: `GET /api/challenge?address={addr}&action=update-pubkey`
- Agent signs challenge and submits with their compressed public key
- Handler verifies the provided pubkey derives to the agent's registered `btcAddress`
- Updates `btcPublicKey` on both `btc:` and `stx:` KV records

**Constraints:**
- Once a valid (non-empty) pubkey is set and verified, there should be no need to change it — the cryptographic binding between L1 and L2 identity is critical
- Consider making this a one-time fix: only allow if current `btcPublicKey` is empty
- Validate the provided pubkey actually derives to the agent's registered BTC address before storing

### Files involved

- `lib/bitcoin-verify.ts` — Source of the empty pubkey (lines 439, 441, 446, 448)
- `app/api/register/route.ts` — Stores it without validation (line 614)
- `lib/challenge.ts` — Add `update-pubkey` handler to `ACTION_HANDLERS` map (line 447-452)
- `lib/types.ts` — `btcPublicKey` field on AgentRecord (line 9)

### PR DESCRIPTION
**Pull request by whoabuddy**
## Summary
- **fix(register)**: Address PR #401 review feedback — use falsy check `!btcResult.publicKey` instead of strict `=== ""`, replace `console.warn` with structured `log.warn()`
- **feat(challenge)**: Add `update-pubkey` challenge action so BIP-322 agents (bc1q/bc1p) can fix their missing `btcPublicKey`. One-time only, validates compressed pubkey derives to registered address. Supports both P2WPKH and P2TR. Auto-populates `nostrPublicKey`. Registration response now includes `btcPublicKeyMissing: true` hint for BIP-322 wallets.
- **refactor(challenge)**: Simplify GitHub username validation to single regex, remove redundant type cast, fix dead code in update-pubkey type guard, use nullish coalescing
- **docs(claude)**: Document all 6 challenge actions in CLAUDE.md

## Context
BIP-322 signature verification (bc1q/bc1p wallets) correctly returns an empty `publicKey` — BIP-322 verifies address ownership but can't recover the key like BIP-137 ECDSA does. The empty string was silently stored, breaking Nostr npub derivation and the L1↔L2 cryptographic binding.

Issue #400 (signal content null) was redirected to aibtcdev/agent-news#67 — wrong repo.

## Test plan
- [ ] Build passes (`npm run build`)
- [ ] Lint passes (`npm run lint`)
- [ ] All 316 tests pass (`npm run test`)
- [ ] Register with BIP-322 wallet — response includes `btcPublicKeyMissing: true` and hint
- [ ] `GET /api/challenge?address={addr}&action=update-pubkey` returns challenge
- [ ] Submit compressed pubkey via challenge — stored on both KV records
- [ ] Reject if pubkey already set (one-time only)
- [ ] Reject if pubkey doesn't derive to registered address

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/challenge.ts [hunk 1]
```diff
@@ -620,6 +619,113 @@ async function handleLinkGitHub(
...
+  const nostrPublicKey = agent.nostrPublicKey ?? xOnlyHex;
+
+  const updated: AgentRecord = {
+    ...agent,
+    btcPublicKey: normalized,
+    nostrPublicKey,
+  };
+
+  return { success: true, updated };
+}
```
[the remark this answers is not in the document]
Comment (whoabuddy) [reply lib/challenge.ts]: Fixed in 06a72f3 — added 12 unit tests covering: P2WPKH success, P2TR success, rejection when btcPublicKey already set, missing/invalid param, wrong format (length, prefix), address mismatch, unsupported address type, and input normalization. Also tests nostrPublicKey auto-population and empty-string overwrite.

**Review by biwasxyz — APPROVED** [review]
Looks good — clean implementation with solid security model.

**Highlights:**
- One-time-only guard + address derivation verification is the right approach for pubkey binding
- Good test coverage (12 tests covering P2WPKH, P2TR, rejection cases, normalization, nostr auto-population)
- Structured logging fix and falsy check address PR #401 feedback nicely
- `validateGitHubUsername` consolidation to single regex is cleaner

**Minor notes (non-blocking):**
- `nostrPublicKey` derivation uses the untweaked x-coordinate (`normalized.slice(2)`), which is correct for nostr (NIP-06 uses untweaked keys), but a brief comment distinguishing this from the tweaked P2TR key would help future readers
- The `btcPublicKeyMissing` / `btcPublicKeyHint` registration response fields don't have test coverage — low risk since it's just a response flag, but could be worth adding later

LGTM 🚀
